app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -54,66 +55,96 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
         $interval.cancel(promise);
     });
 
-    $scope.$on("render", function(ev, vertices, added) {
+    $scope.$on('render', function(ev, vertices, added) {
         /*
          * We are passed two selections of <g> elements:
          * vertices: All the elements
          * added: Just the ones that were added
          */
-        added.attr("class", function(d) { return d.item.kind; });
-        added.append("circle")
-            .attr("r", function(d) { return self.getDimensions(d).r})
+        added.attr('class', function(d) {
+            return d.item.kind;
+        });
+        added.append('circle')
+            .attr('r', function(d) {
+                return self.getCircleDimensions(d).r;
+            })
             .attr('class' , function(d) {
-              return topologyService.getItemStatusClass(d);
+                return topologyService.getItemStatusClass(d);
             });
-        added.append("title");
-        added.on("dblclick", function(d) {
-            return self.dblclick(d);});
-        added.append("image")
-            .attr("xlink:href",function(d) {
-                return "/assets/svg/" + self.class_name(d) + ".svg";
+        added.append('title');
+        added.on('dblclick', function(d) {
+          return self.dblclick(d);
+        });
+        added.append('image')
+            .each(function(d) {
+                var iconInfo = self.getIcon(d);
+                if (iconInfo.type != 'image') {
+                    return;
+                }
+                $(this).attr('href', iconInfo.icon)
+                .attr('y', self.getCircleDimensions(d).y)
```

Comment:
@Jiri-Kremser thanks :)

As for ES6, it is possible.. with the limitation that it will go through babel5 and the file needs to be `*.es6` not `*.js`. (And the specs do need to be ES5 for now.)

We should hopefully get the tooling up to date before the Euwe release..